### PR TITLE
Post an event when a connection changes to ZeroRtt state.

### PR DESCRIPTION
**Pull request by ddragana**
Also add a funcion is_zerortt().

### CODE REVIEW

**Comment by ddragana** [comment]
We need this for necko. I am not sure which one will be better to use he event or the is_zerortt(), I need to test it a bit. The event will work for sure, but the function may make some things faster but it may be tricky because on the order of events... I just need to test it a bit.

File: neqo-http3/src/client_events.rs [hunk 1]
```diff
@@ -184,6 +184,9 @@ impl Http3ClientEvents {
         // If closing, existing events no longer relevant.
```
Comment (martinthomson) [inline neqo-http3/src/client_events.rs:184]: Fix the commentary or move it.

File: neqo-http3/src/connection_client.rs [hunk 2]
```diff
@@ -3246,6 +3253,9 @@ mod tests {
             .expect("Set resumption token.");
 
         assert_eq!(client.state(), Http3State::ZeroRtt);
+        let zerortt_event = |e| matches!(e, Http3ClientEvent::StateChange(Http3State::ZeroRtt));
+        assert!(client.events().any(zerortt_event));
+        assert!(client.is_zerortt());
```
Comment (martinthomson) [inline neqo-http3/src/connection_client.rs:3258]: How does `is_zerortt` differ from `state() == Http3State::ZeroRtt` ?

File: neqo-http3/src/client_events.rs [hunk 3]
```diff
@@ -184,6 +184,9 @@ impl Http3ClientEvents {
...
         match state {
             Http3State::Closing { .. } | Http3State::Closed(_) => self.events.borrow_mut().clear(),
+            Http3State::Connected => {
+                self.remove(|evt| matches!(evt, Http3ClientEvent::StateChange(Http3State::ZeroRtt)))
```
Comment (martinthomson) [inline neqo-http3/src/client_events.rs:188]:
```suggestion
                self.remove(|evt| matches!(evt, Http3ClientEvent::StateChange(Http3State::ZeroRtt)));
```

File: neqo-http3/src/connection_client.rs (hunk 2 above)
Comment (ddragana) [reply neqo-http3/src/connection_client.rs:3258]: Ok, I can remove is_zerortt. I can implement it in neqo_glue code.

**Pull request merged**